src/cranbtree.c

```diff
@@ -201,6 +201,10 @@ void *cbt_update(cranbtree_t * bt, int key, void *object)
 
 	void *old_object = cbt_update_helper(bt->root, key, object, bt->n);
 
+	if(cbt_update_if_exists(bt, key, old_object) == NULL){
```

Comment:
The object is already updated in cbt_update_helper in the line above. No need to reupdate again. instead you should check if `old_object` is NULL instead of checking if cbt_update_if_exists is NULL. It passes the tests for now becuase cbt_update_if_exists is not yet implemented and always returns NULL. so this condition is always satisfied